src/main/webapp/app/exercises/shared/result/result-detail.component.html

```diff
@@ -75,6 +75,14 @@ <h4 jhiTranslate="artemisApp.result.buildError">Build Error:</h4>
     <ng-template #noFeedbacksContainer>
         <pre jhiTranslate="artemisApp.result.noResultDetails">No result details available</pre>
     </ng-template>
+    <!--
+    Show and link to the commit hash in case of programming exercise results
+ -->
+    <p *ngIf="exerciseType === ExerciseType.PROGRAMMING && result.submission">
+        <span>{{ 'artemisApp.result.linkedCommit' | artemisTranslate }}</span>
+        <a *ngIf="getCommitUrl(); else commitWithoutLink" href="{{ getCommitUrl() }}" target="_blank" rel="noopener noreferrer">{{ getCommitHash().substr(0, 11) }}</a>
```

Comment:
I think these function calls could be pre-calculated with an attribute (Client guidelines Section 0)

Reply:
Yes, we can do that, it is quite unlikely they will change.